crates/nu-protocol/src/engine/engine_state.rs

```diff
@@ -2420,4 +2452,39 @@ mod engine_state_tests {
 
         Ok(())
     }
+
+    #[test]
+    fn test_find_non_whitespace_index() {
+        let commands = vec![
+            ("    hello", 4, false),
+            ("    sudo ", 4, true),
+            ("sudo ", 0, true),
+            ("	hello", 1, false),
+            ("	sudo", 1, false),
+            ("	sudo ", 1, true),
+            (" 	sudo ", 2, true),
+            ("	 sudo ", 2, true),
+            ("	hello ", 1, false),
+            ("    hello | sudo ", 4, true),
+            ("    sudo|sudo", 4, false),
+            ("sudo | sudo ", 0, true),
+            ("	hello sud", 1, false),
+            ("	sudo | sud ", 1, false),
+            ("	sudo|sudo ", 1, true),
+            (" 	sudo | sudo ls | sudo ", 2, true),
+        ];
```

Comment:
Maybe I am wrong but this feels like two independent tests in one. 
The test name says it tests your string trimming helper but it also contains code specific to the `sudo` goal. For the former it feels like overtesting but that might not be the case for the latter.